packages/appstore/src/components/modals/wallet-modal/wallet-modal.scss

```diff
@@ -8,91 +8,158 @@
+            visibility: visible;
+            opacity: 1;
+            transition: visibility 0s, opacity 0.2s, height 0.2s ease;
+        }
+    }
 
-            .dc-mobile-dialog__wallet-mobile-dialog__wrapper {
-                .modal-header {
-                    height: 12.2rem;
-                    transition: height 0.2s ease;
-
-                    .title-visibility {
-                        height: 2rem;
-                    }
-
-                    .title-visibility,
-                    .icon-visibility {
-                        visibility: visible;
-                        opacity: 1;
-                        transition: visibility 0s, opacity 0.2s, height 0.2s ease;
-                    }
-                }
+    //TODO: check do we need this after bg change to radial-gradient
+    &__title-wrapper {
+        position: relative;
+    }
+
+    &__title {
+        display: flex;
+        align-items: center;
+
+        &-wallet {
+            padding-right: 0.8rem;
+            color: var(--text-general);
+
+            &-demo {
+                color: var(--demo-text-color-1);
             }
+        }
 
-            .scrolled-content {
-                .modal-header {
-                    height: 8.2rem;
+        &-balance {
+            color: var(--text-prominent);
 
-                    .title-visibility,
-                    .icon-visibility {
-                        visibility: hidden;
-                        opacity: 0;
-                        height: 0;
-                    }
-                }
+            &-demo {
+                color: var(--demo-text-color-2);
             }
         }
     }
+
+    &--hidden-title {
+        height: 8.2rem;
+        align-items: center;
+        justify-content: space-between;
+
+        .title-visibility,
+        .icon-visibility {
+            visibility: hidden;
+            opacity: 0;
+            height: 0;
+        }
+    }
+
+    &__currency-icon {
+        z-index: 3;
+        margin-left: auto;
+        margin-right: 1.6rem;
+
+        @include mobile {
+            margin-right: 0.8rem;
+        }
+    }
+
+    &__close-icon {
+        position: relative;
+
+        .dc-icon {
+            cursor: pointer;
+        }
+    }
 }
 
 // styles for mobile and desktop modal body (tabs with content)
 .dc-tabs--modal-body__tabs {
-    max-width: 128rem;
     width: 100%;
     margin-top: -4.8rem;
 
     @include mobile {
         top: 8.2rem;
         margin-top: 0;
+        transition: top 0.2s ease;
+
+        &.is_scrolled {
+            top: 4.2rem;
+        }
+    }
+
+    &__themed-scrollbar {
+        width: 100%;
+    }
+
+    &__content-wrapper {
+        max-width: 128rem;
+        margin: 0 auto;
+        padding: 2.4rem 4rem;
+
+        @include mobile {
+            padding: 1.6rem;
+        }
     }
 
     .dc-tabs__list {
         padding: 0 4rem;
 
         @include mobile {
             padding: 0 1.6rem;
+        }
+
+        &--header--modal-body__tabs {
```

Comment:
Can we have a better class name? it's a modifier and block and element at the same time :)

something like: `&-header`

Reply:
yeah @hamid-deriv it looks weird, but unfortunately this implementation is in `Tabs` component itself and we need to refactor `Tabs` component for proper classname visibility